dds/DCPS/RTPS/Spdp.cpp

```diff
@@ -4211,6 +4220,17 @@ ParticipantData_t& Spdp::get_participant_data(const DCPS::RepoId& guid)
   return iter->second.pdata_;
 }
 
+DCPS::MonotonicTime_t Spdp::get_participant_discovered_at() const
+{
+  return participant_discovered_at_;
+}
+
+DCPS::MonotonicTime_t Spdp::get_participant_discovered_at(const DCPS::RepoId& guid) const
+{
+  DiscoveredParticipantConstIter iter = participants_.find(make_part_guid(guid));
+  return iter->second.discovered_at_;
```

Comment:
Does this need to handle the case where it was not found?

Reply:
It is parallel to `Spdp::get_participant_data` so I don't believe so.